### DISCUSSION TITLE
Number formatting functions?

### DISCUSSION THREAD

**Discussion by qrilka** [sharkdp/numbat#669] [Q&A] [answered]
From my calculations I have a number with a bunch of zeroes which gets shown in scientific notation - is there any way to format it with digits only?

**Comment by sharkdp** [comment, marked as answer]
We currently lack some proper functionality for formatting Numbers. See also this ticket: https://github.com/sharkdp/numbat/issues/219

One workaround to do this right now is to format it as a string. For example:
```
>>> let x = 1e20
>>> x

    = 1.0e+20

>>> "{x:.0}"

    = "100000000000000000000"    [String]
```

You could even make a function out of this and add it to your `init.nbt` file:
```
fn format_normal(q) = "{q:.0}"
```

Then you do something like
```
>>> x -> format_normal

    = "100000000000000000000"    [String]
```

That said, it's still just a workaround. We should support this natively. Please feel free to comment in the linked ticket if you have additional ideas or like to contribute in another way.
